src/ir/context.rs

```diff
@@ -2412,6 +2428,70 @@ If you encounter an error missing from this list, please file an issue or a PR!"
         }
     }
 
+    /// Compute the items for dynamic generation.
+    fn compute_dynamic_generation_items(&mut self) {
+        assert!(self.in_codegen_phase());
+        assert!(self.current_module == self.root_module);
+        let _t = self.timer("compute_dynamic_generation_items");
+
+        let dyngen_items = self
+            .items()
+            // Only consider items that are enabled for codegen.
+            .filter(|&(_, item)| item.is_enabled_for_codegen(self))
+            .filter(|&(_, item)| {
+                // If the user has not chosen to do dynamic loading, then we have nothing to
+                // do.
+                if !self.options().dynamic_loading {
+                    return false;
+                }
+
+                let name = item.path_for_whitelisting(self)[1..].join("::");
+
+                // If there is a whitelist and this function is not on it, don't add it.
+                if !self.options().whitelisted_functions.is_empty() &&
+                    !self.options().whitelisted_functions.matches(&name)
+                {
```

Comment:
I would expect this to be accounted for via the regular whitelisting mechanism, why is this needed?

Reply:
Yep, I think this can be simplified. I think I had the wrong idea of the relationship between `items`, `codegen_items`, and `whitelisted_items`. I've changed this to iterate through `self.items()`, but filter out anything whose item ID is not inside `codegen_items`. It seems to do the job -- is that what you had in mind?